index.js

```diff
@@ -11,7 +11,7 @@ var slice = Array.prototype.slice;
 
 var boundFindShim = function find(array, predicate) {
 	ES.RequireObjectCoercible(array);
-	return implementation.call(array, predicate);
```

Comment:
I think this would be better as `implementation.apply(array, arguments)`. Indexing into arguments objects (or arrays) is generally not a good idea.